pallets/thea/src/lib.rs

```diff
@@ -502,8 +473,146 @@ pub mod pallet {
 			T::TheaPalletId::get().into_account_truncating()
 		}
 
+		pub fn do_deposit(
+			token_type: TokenType,
+			payload: Vec<u8>,
+			bit_map: u128,
+			bls_signature: [u8; 96],
+		) -> Result<(), DispatchError> {
+			let approved_deposit = Self::router(token_type, payload.clone())
+				.map_err(|_| Error::<T>::NoApprovedDeposit)?;
+			let current_active_relayer_set =
+				Self::get_relayers_key_vector(approved_deposit.network_id);
+
+			ensure!(
+				thea_primitives::thea_ext::bls_verify(
+					&bls_signature,
+					bit_map,
+					&payload,
+					&current_active_relayer_set.into_inner()
+				),
+				Error::<T>::BLSSignatureVerificationFailed
+			);
+			<DepositNonce<T>>::insert(
+				approved_deposit.network_id.saturated_into::<Network>(),
+				approved_deposit.deposit_nonce + 1,
+			);
+
+			if <ApprovedDeposits<T>>::contains_key(&approved_deposit.recipient) {
+				<ApprovedDeposits<T>>::mutate(approved_deposit.recipient.clone(), |bounded_vec| {
+					if let Some(inner_bounded_vec) = bounded_vec {
+						inner_bounded_vec.try_push(approved_deposit.clone()).unwrap();
+					}
+				});
+			} else {
+				let mut my_vec: BoundedVec<ApprovedDeposit<T::AccountId>, ConstU32<100>> =
+					Default::default();
+				if let Ok(()) = my_vec.try_push(approved_deposit.clone()) {
+					<ApprovedDeposits<T>>::insert::<
+						T::AccountId,
+						BoundedVec<ApprovedDeposit<T::AccountId>, ConstU32<100>>,
+					>(approved_deposit.recipient.clone(), my_vec);
+					<AccountWithPendingDeposits<T>>::mutate(|accounts| {
+						accounts.insert(approved_deposit.recipient.clone())
+					});
+				}
+			}
```

Comment:
After moving here - todo:
> // TODO[#610]: Will be refactored in integration with Staking

was lost. Referenced [issue](https://github.com/Polkadex-Substrate/Polkadex/issues/610) was closed 2 weeks ago by [this](https://github.com/Polkadex-Substrate/Polkadex/pull/615) PR. @nakul1010 please take into account and fix it.